src/library_pthread.js

```diff
@@ -983,11 +983,6 @@ var LibraryPThread = {
     if (!ENVIRONMENT_IS_PTHREAD) _exit(status);
     else PThread.threadExit(status);
     // pthread_exit is marked noReturn, so we must not return from it.
-    if (ENVIRONMENT_IS_NODE) {
-      // exit the pthread properly on node, as a normal JS exception will halt
-      // the entire application.
-      process.exit(status);
-    }
```

Comment:
Do you know why this was here in the first place?  I don't really understand the comment or why  process.exit would ever maskde sense for exiting a thread.  (unless perhaps process.exit only applies to the current worker thread which seem unlikely).

In order works I agree this change looks good I just can't see why it was ever any different.  Perhaps @kripken can lend some background.   I believe this code dates back to the very first pthread node code: #9745

Also, can you mention this fix in the title and/or PR description so it describes what you are fixing exacly.

Reply:
This change was discovered by trial-and-error as a attempt to fix the failing Node posix tests. So, I'm not entirely sure why it was added (with commit https://github.com/emscripten-core/emscripten/pull/9745/commits/5a8c86db5d955d45ff54b6b73d7d02e79b314f95) and whether this change is correct.

I'll fix the title and PR description, after I've investigated it a bit further.

Reply:
Is adding EXIT_RUNTIME enough to make this test keep working?

Reply:
After some further research, it looks like the `process.exit()` call in a Node worker will immediately cause the thread to exit, and not the whole program. Somehow this call also prevents from new threads to be created(?) and causes the program to hang indefinitely (see for example the added `test_pthread_exit.c` test case in https://github.com/kleisauke/emscripten/commit/e21afb7dc135f162b69c78bc377891580254c4c1 which hangs on master). The failing stress-test mentioned in #9763 now also seems to work correctly on Node.

`EXIT_RUNTIME` doesn't make the `test_pthread_create` test case pass after this change, but calling explicitly `exit(0);` after cancelling the main loop does (see commit https://github.com/kleisauke/emscripten/commit/fb35f29f4cc614db2770de9e54b529c925dfc257).

Reply:
I think I added this because indeed the throw on the next line would shut down the entire program (threads + main runtime), as the comment says. And `process.exit` seemed to just shut down the thread, but I can't find any docs to justify that, so it may have been a reliance on undefined behavior...

But this testcase generally shows that:
```js
const { Worker, isMainThread, parentPort } = require('worker_threads');
if (isMainThread) {
  // This code is executed in the main thread and not in the worker.
  
  // Create the worker.
  const worker = new Worker(__filename);

  setInterval(() => { console.log('time') }, 1000);
} else {
  // This code is executed in the worker and not in the main thread.

  process.exit();
  //throw "foo";
}
```
Replace the process.exit with the throw to see the result.

But, it does look good to me to remove the process.exit here, assuming the throw does not halt the entire program. Do we have tests checking that, that is, that a node pthread can exit while things keep running?

Reply:
(If things work then I assume we catch that throw higher up.)